{"context": "\n\n\u74b0\u5883\n\u30fbCentOS7\n\u30fbpyenv + Anaconda\n\n\u53c2\u8003\n[pyenv]https://github.com/yyuu/pyenv\n\n\u5c0e\u5165\nsudo yum install git -y\n\ngit clone https://github.com/yyuu/pyenv.git ~/.pyenv\n\necho 'export PYENV_ROOT=\"$HOME/.pyenv\"' >> ~/.bash_profile\necho 'export PATH=\"$PYENV_ROOT/bin:$PATH\"' >> ~/.bash_profile\necho 'eval \"$(pyenv init -)\"' >> ~/.bash_profile\n\nsource ./.bash_profile\n\n\nAnaconda\n\n\u5c0e\u5165\u3067\u304d\u308bAnaconda\u3092\u78ba\u8a8d\npyenv install --list | grep anaconda\n\n\nAnaconda\u3092\u5c0e\u5165\npyenv install anaconda3-4.1.1\n\n\nPython2\u7cfb=anaconda2~\nPython3\u7cfb=anaconda3~\n\n\nError\u306b\u306a\u3063\u305f\nDownloading Anaconda3-4.1.1-Linux-x86_64.sh...\n-> https://repo.continuum.io/archive/Anaconda3-4.1.1-Linux-x86_64.sh\nInstalling Anaconda3-4.1.1-Linux-x86_64...\n\nBUILD FAILED (CentOS Linux 7 using python-build 1.0.6-13-g4d96271)\n\nInspect or clean up the working tree at /tmp/python-build.20170114112223.16911\nResults logged to /tmp/python-build.20170114112223.16911.log\n\nLast 10 log lines:\n/tmp/python-build.20170114112223.16911 ~\n/tmp/python-build.20170114112223.16911/Anaconda3-4.1.1-Linux-x86_64 /tmp/python-build.20170114112223.16911 ~\nPREFIX=/home/centos/.pyenv/versions/anaconda3-4.1.1\ntar (child): bzip2: exec \u4e0d\u80fd: \u305d\u306e\u3088\u3046\u306a\u30d5\u30a1\u30a4\u30eb\u3084\u30c7\u30a3\u30ec\u30af\u30c8\u30ea\u306f\u3042\u308a\u307e\u305b\u3093\ntar (child): Error is not recoverable: exiting now\ntar: Child returned status 2\ntar: Error is not recoverable: exiting now\n\n\nError\u306e\u539f\u56e0\u3089\u3057\u304dbzip2\u3092\u5c0e\u5165\nsudo yum install bzip2 -y\n\n\n\u518d\u5ea6Anaconda\u3092\u5c0e\u5165\npyenv install anaconda3-4.1.1\n\n\n\u3046\u307e\u304f\u3044\u3063\u305f\n$pyenv versions\n* system (set by /home/centos/.pyenv/version)\n  anaconda3-4.1.1\n\n\nPython\u306e\u5c0e\u5165\npyenv install 3.6.0\n\n\nError\u306b\u306a\u3063\u305f\nDownloading Python-3.6.0.tar.xz...\n-> https://www.python.org/ftp/python/3.6.0/Python-3.6.0.tar.xz\nInstalling Python-3.6.0...\n\nBUILD FAILED (CentOS Linux 7 using python-build 1.0.6-13-g4d96271)\n\nInspect or clean up the working tree at /tmp/python-build.20170114122724.27883\nResults logged to /tmp/python-build.20170114122724.27883.log\n\nLast 10 log lines:\nchecking for --with-universal-archs... no\nchecking MACHDEP... linux\nchecking for --without-gcc... no\nchecking for --with-icc... no\nchecking for gcc... no\nchecking for cc... no\nchecking for cl.exe... no\nconfigure: error: in `/tmp/python-build.20170114122724.27883/Python-3.6.0':\nconfigure: error: no acceptable C compiler found in $PATH\nSee `config.log' for more details\n\n\nGCC\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\nsudo yum install gcc -y\n\n\n\u307e\u3060Error\nDownloading Python-3.6.0.tar.xz...\n<\u4e2d\u7565>\nzipimport.ZipImportError: can't decompress data; zlib not available\nmake: *** [install] \u30a8\u30e9\u30fc 1\n\n\nZlib\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\nsudo yum install zlib -y\n\n\n\u7d50\u679c\u3001\u3053\u306e\u8fba\u3092\u5bfe\u5fdc\u3057\u306a\u3044\u3068\u3044\u3051\u306a\u304b\u3063\u305f\u3002\nhttps://github.com/yyuu/pyenv/wiki\n\nCentOS/Fedora 21 and below:\nFIXME: you may need to install xz to build some CPython version\n\nyum install gcc zlib-devel bzip2 bzip2-devel readline-devel sqlite sqlite-devel openssl-devel\n\n# \u74b0\u5883\n\u30fbCentOS7\n\u30fbpyenv + Anaconda\n\n\n# \u53c2\u8003\n[pyenv]https://github.com/yyuu/pyenv\n\n# \u5c0e\u5165 \n\n```\nsudo yum install git -y\n```\n\n```\ngit clone https://github.com/yyuu/pyenv.git ~/.pyenv\n```\n\n```\necho 'export PYENV_ROOT=\"$HOME/.pyenv\"' >> ~/.bash_profile\necho 'export PATH=\"$PYENV_ROOT/bin:$PATH\"' >> ~/.bash_profile\necho 'eval \"$(pyenv init -)\"' >> ~/.bash_profile\n```\n\n```\nsource ./.bash_profile\n```\n\n##Anaconda\n\n### \u5c0e\u5165\u3067\u304d\u308bAnaconda\u3092\u78ba\u8a8d\n```\npyenv install --list | grep anaconda\n```\n###Anaconda\u3092\u5c0e\u5165\n```\npyenv install anaconda3-4.1.1\n```\n> Python2\u7cfb=anaconda2~\n> Python3\u7cfb=anaconda3~\n\n\n###Error\u306b\u306a\u3063\u305f\n```\nDownloading Anaconda3-4.1.1-Linux-x86_64.sh...\n-> https://repo.continuum.io/archive/Anaconda3-4.1.1-Linux-x86_64.sh\nInstalling Anaconda3-4.1.1-Linux-x86_64...\n\nBUILD FAILED (CentOS Linux 7 using python-build 1.0.6-13-g4d96271)\n\nInspect or clean up the working tree at /tmp/python-build.20170114112223.16911\nResults logged to /tmp/python-build.20170114112223.16911.log\n\nLast 10 log lines:\n/tmp/python-build.20170114112223.16911 ~\n/tmp/python-build.20170114112223.16911/Anaconda3-4.1.1-Linux-x86_64 /tmp/python-build.20170114112223.16911 ~\nPREFIX=/home/centos/.pyenv/versions/anaconda3-4.1.1\ntar (child): bzip2: exec \u4e0d\u80fd: \u305d\u306e\u3088\u3046\u306a\u30d5\u30a1\u30a4\u30eb\u3084\u30c7\u30a3\u30ec\u30af\u30c8\u30ea\u306f\u3042\u308a\u307e\u305b\u3093\ntar (child): Error is not recoverable: exiting now\ntar: Child returned status 2\ntar: Error is not recoverable: exiting now\n```\n\n###Error\u306e\u539f\u56e0\u3089\u3057\u304dbzip2\u3092\u5c0e\u5165\n```\nsudo yum install bzip2 -y\n```\n###\u518d\u5ea6Anaconda\u3092\u5c0e\u5165\n```\npyenv install anaconda3-4.1.1\n```\n\n###\u3046\u307e\u304f\u3044\u3063\u305f\n\n```\n$pyenv versions\n* system (set by /home/centos/.pyenv/version)\n  anaconda3-4.1.1\n```\n\n###Python\u306e\u5c0e\u5165\n```\npyenv install 3.6.0\n```\n\n###Error\u306b\u306a\u3063\u305f\n```\nDownloading Python-3.6.0.tar.xz...\n-> https://www.python.org/ftp/python/3.6.0/Python-3.6.0.tar.xz\nInstalling Python-3.6.0...\n\nBUILD FAILED (CentOS Linux 7 using python-build 1.0.6-13-g4d96271)\n\nInspect or clean up the working tree at /tmp/python-build.20170114122724.27883\nResults logged to /tmp/python-build.20170114122724.27883.log\n\nLast 10 log lines:\nchecking for --with-universal-archs... no\nchecking MACHDEP... linux\nchecking for --without-gcc... no\nchecking for --with-icc... no\nchecking for gcc... no\nchecking for cc... no\nchecking for cl.exe... no\nconfigure: error: in `/tmp/python-build.20170114122724.27883/Python-3.6.0':\nconfigure: error: no acceptable C compiler found in $PATH\nSee `config.log' for more details\n```\n\n###GCC\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\n```\nsudo yum install gcc -y\n```\n\n###\u307e\u3060Error\n```\nDownloading Python-3.6.0.tar.xz...\n<\u4e2d\u7565>\nzipimport.ZipImportError: can't decompress data; zlib not available\nmake: *** [install] \u30a8\u30e9\u30fc 1\n```\n\n###Zlib\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\n```\nsudo yum install zlib -y\n```\n### \u7d50\u679c\u3001\u3053\u306e\u8fba\u3092\u5bfe\u5fdc\u3057\u306a\u3044\u3068\u3044\u3051\u306a\u304b\u3063\u305f\u3002\nhttps://github.com/yyuu/pyenv/wiki\n\n> CentOS/Fedora 21 and below:\n> FIXME: you may need to install xz to build some CPython version\n\n```\nyum install gcc zlib-devel bzip2 bzip2-devel readline-devel sqlite sqlite-devel openssl-devel\n```\n", "tags": ["centos7", "Anaconda"]}